### PR TITLE
[algo] fix: micro-batch normalization for distillation loss

### PR DESCRIPTION
**Pull request by JacobHelwig**
### What does this PR do?

Applies fix proposed by @wuxibin89 for issue #7200 found by @yph22

### Test

image

- Ran Qwen3-0.6B student w/ Qwen3-1.7B teacher on gsm8k for a fixed batch using `RolloutSkip`
- `fixOff` == prior to this PR, `fixOn` == with this PR
- Total batch size for all 4 is 8, `mb{k}` == mico-batch size of size `k` (`mb1` --> 8 accumulations, `mb8` == 1 accumulations)
- `Off` with mb=1 shows 8x higher loss, after applying fix, loss is correctly normalized and is invariant to the micro-batch size 

#### PG branch

With pg enabled (suggested by @ErenAta16)

image

### CODE REVIEW

**Review by ErenAta16 — COMMENTED** [review]
The normalization values look right and the before/after curves are the kind of evidence this needs. One question about placement, because the fix lands inside one of the two branches and the other one reads the same dict earlier.

`distillation_loss` splits on `loss_config.use_policy_gradient`:

```python
if loss_config.use_policy_gradient:
    policy_loss_fn = get_policy_loss_fn(loss_config.policy_loss_mode)
    for k, v in config.global_batch_info.items():          # copies whatever is present at this point
        loss_config.global_batch_info[k] = v
    ...
    distillation_loss, pg_metrics = policy_loss_fn(..., config=loss_config, ...)
else:
    # Directly backpropagate distillation loss as a supervised loss
    ...
    config.global_batch_info["dp_size"] = data["dp_size"]  # <- this PR
    config.global_batch_info["batch_num_tokens"] = ...
    distillation_loss = agg_loss(..., **config.global_batch_info)
```

The four assignments sit after the `else`, so they only run on the supervised path. The policy-gradient path copies `config.global_batch_info` into `loss_config` further up and never sees them.

That leaves two possibilities and I cannot tell which from this file alone:

If `config.global_batch_info` is already populated by the caller before `distillation_loss` runs, then the copy in the PG branch is fine and these four lines are re-stating values that were already there, in which case the supervised path was reading them too and there would have been nothing to fix. The fact that this changes the curves suggests that is not the case.

If it is not populated by the caller, which the benchmark implies, then the PG branch copies an empty or stale dict and `use_policy_gradient=True` runs still lose the micro-batch normalization after this PR. That would make the fix cover one of the two distillation modes.

Which mode did the gsm8k run use? If it was the supervised path, the PG path is worth a second look before this lands, since #7200 does not read as mode-specific. Hoisting the four assignments above the `if` would cover both and would not change the supervised behaviour you measured.

Not asserting it is broken; the evidence I have is the branch structure, and the caller side is where it resolves.

**Comment by JacobHelwig** [comment]
Thanks @ErenAta16 , you're right. Applied the fix and updated the tests

**Review by wuxibin89 — APPROVED** [review]
